Upgrade tests can't find cluster, report SUCCESS anyway

```
10:34:50 2015/12/04 10:34:50 e2e.go:309: Error running get status: exit status 1
10:34:50 2015/12/04 10:34:50 e2e.go:305: Step 'get status' finished in 287.79716ms
10:34:50 2015/12/04 10:34:50 e2e.go:271: Testing requested, but e2e cluster not up!
...
10:35:08 Finished: SUCCESS
```

If we are uploading JUnit results, this situation will fail because it can't find test results.  It should fail regardless of whether JUnit results are being uploaded, though.

--- a/hack/jenkins/e2e.sh
+++ b/hack/jenkins/e2e.sh
@@ -1477,6 +1477,8 @@ fi
 # Jenkins will look at the junit*.xml files for test failures, so don't exit
 # with a nonzero error code if it was only tests that failed.
 if [[ "${E2E_TEST,,}" == "true" ]]; then
+    # Check to make sure the cluster is up before running tests, and fail if it's not.
+    go run ./hack/e2e.go ${E2E_OPT} -v --isup
     go run ./hack/e2e.go ${E2E_OPT} -v --test --test_args="${GINKGO_TEST_ARGS}" && exitcode=0 || exitcode=$?
     if [[ "${E2E_PUBLISH_GREEN_VERSION:-}" == "true" && ${exitcode} == 0 && -n ${githash:-} ]]; then
         echo "publish githash to ci/latest-green.txt: ${githash}"

